models/classes/Platform/Service/AccessTokenRequestValidator.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoLti\models\classes\Platform\Service;
+
+use OAT\Library\Lti1p3Core\Registration\RegistrationRepositoryInterface;
+use OAT\Library\Lti1p3Core\Service\Server\Validator\AccessTokenRequestValidationResult;
+use OAT\Library\Lti1p3Core\Service\Server\Validator\AccessTokenRequestValidator as Lti1p3AccessTokenRequestValidator;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoLti\models\classes\Platform\Repository\Lti1p3RegistrationRepository;
+use Psr\Http\Message\ServerRequestInterface;
+
+class AccessTokenRequestValidator extends ConfigurableService
+{
+    public function validate(ServerRequestInterface $request, string $role): AccessTokenRequestValidationResult
```

Comment:
```suggestion
    /**
     * @throws XXXXXXXX
     */
    public function validate(ServerRequestInterface $request, string $role): void
```

You are exposing external library object on this extension and other, so you are coupling this library to the internal services interfaces. 

Please, as a validator, throw the exception with proper message instead. It is up to the client to handle this exception in the right way.